### ISSUE TITLE
Gameplay | Timer support

### ISSUE THREAD

**Issue by Hill-98** [Hastwell/Omukade.Cheyenne#8]
Currently, the countdown in the game is locked at 39 minutes. Is this impossible to achieve, or has it not been implemented yet?

Because this may cause one player to maliciously delay the game.

**Comment by Hastwell** [comment]
This is merely not implemented, and was not an immediate priority. The intended use case of Cheyenne is casual play with a group or community of people you already know, and you trust to not abuse this behavior.

Cheyenne transmits that timers are disabled for matches, although this is possibly exposing a bug in the vanilla client as it was likely not designed or tested to handle ladder matches without a timer.

**Comment by Junglemania** [comment]
> This is merely not implemented, and was not an immediate priority. The intended use case of Cheyenne is casual play with a group or community of people you already know, and you trust to not abuse this behavior.
> 
> Cheyenne transmits that timers are disabled for matches, although this is possibly exposing a bug in the vanilla client as it was likely not designed or tested to handle ladder matches without a timer.

Well, we have a lot of players, thousands of them, and the number of players is still growing rapidly, so there will always be people who don't follow the rules.

**Comment by Hill-98** [comment]
```C#
new PlayerSettings { gameMode = GameMode.Standard, gameplayType = GameplayType.Ranked, useAutoSelect = false, useMatchTimer = true, useOperationTimer = true, matchMode = MatchMode.Standard, matchTime = 1500f };
```

I noticed that I only need to change useMatchTimer and useOperationTimer to true to enable the timers, but it doesn't do that by default, what's wrong with that?

**Comment by Hill-98** [comment]
I implemented simple timer support, currently in testing on my community server, and will create a PR if there are no issues after a while.

**Comment by Hastwell** [comment]
Beta timer support has been implemented in 673ce0a3ed8f47a9c8c3b65acb4bee6bc0db8a50. It may be enabled with the `config.json` directive `"enable-game-timers": true`. The server performs no enforcement of timers; it is up to clients to determine when the allowed time has elapsed and trigger appropriate behavior.
